docs/modules/configuration/pages/jet-configuration.adoc

```diff
@@ -99,6 +99,11 @@ HazelcastInstance instance = createHazelcastInstance(config);
 
 By default, resource uploading for Jet engine jobs is disabled.
 
+[%hardbreaks]
+WARNING: Jet job's classloaders are destroyed after the job is completed. Please have in mind, that asynchronous cleanup and finalizers do not block job completion as other thread(s) would be executing those finalization/cleanup steps.
+If there is some asynchronous cleanup in attached classes or some class use finalization mechanism, it may try to load some classes after the classloaders are destroyed, resulting in Class Loader leak.
```

Comment:
this can lead not only to classloader leak, but also other memory leaks due to failed finalizers or async cleanup

Reply:
done by Rob in https://github.com/hazelcast/hz-docs/pull/1971/commits/a1e74b533bc26e08601d8638dbe3ecf454e05f58